pkg/kubelet/nodestatus/setters.go

```diff
@@ -126,8 +126,8 @@ func NodeAddress(nodeIPs []net.IP, // typically Kubelet.nodeIPs
 
 				nodeIPTypes := make(map[v1.NodeAddressType]bool)
 				for _, nodeAddress := range cloudNodeAddresses {
-					if nodeAddress.Address == nodeIP.String() {
-						enforcedNodeAddresses = append(enforcedNodeAddresses, v1.NodeAddress{Type: nodeAddress.Type, Address: nodeAddress.Address})
+					if net.ParseIP(nodeAddress.Address).Equal(nodeIP) {
```

Comment:
```suggestion
					if  netutils.ParseIPSloppy(nodeAddress.Address).Equal(nodeIP) {
```

we need to use this forked parsers beacuse of the change on the golang parsers behavior on 1.17

Reply:
makes sense. will do

Reply:
the CI blocks using the new parsers :)  https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/107736/pull-kubernetes-verify/1485707280572747776